### PR TITLE
Feat: [Term Entry] PyTorch Tensor Operations .index_copy_()

### LINKED ISSUE #5615: [Term Entry] PyTorch Tensor Operations .index_copy()
### Reason/inspiration (optional)

We would like a new entry on the `.index_copy_()` term under PyTorch. The entry should go in a new file under `docs/content/pytorch/concepts/tensor-operations/terms/index-copy/index-copy.md`.

The entry should include:

- An introduction to the concept
- A `Syntax` section that provides the syntax for the concept
- An `Example` section that provides an example demonstrating the concept in use

Please refer to the [term entry template](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md) and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by SaiTeja-002**
### Description

I've added a new entry for `.index_copy_()` in PyTorch at `docs/content/pytorch/concepts/tensor-operations/terms/index-copy/index-copy.md`

### Issue Solved

closes #5615 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/pytorch/concepts/tensor-operations/terms/index-copy/index-copy.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+---
+Title: 'index_copy()'
+Description: 'Returns a new tensor by copying values to specified indices of a given tensor along the specified dimension.'
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/index-copy/index-copy.md:3]:
```suggestion
Description: 'Returns a new tensor by copying values from a source tensor to specified indices in the target tensor along a given dimension.'
```

Comment (mamtawardhani) [reply content/pytorch/concepts/tensor-operations/terms/index-copy/index-copy.md:3]: reframed for better readability

File: content/pytorch/concepts/tensor-operations/terms/index-copy/index-copy.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  - 'Python'
+  - 'Machine Learning'
+  - 'Methods'
+  - 'Functions'
+CatalogContent:
+  - 'intro-to-py-torch-and-neural-networks'
+  - 'paths/computer-science'
+---
+
+The **`index_copy()`** function in PyTorch creates a new [tensor](https://www.codecademy.com/resources/docs/pytorch/tensors) with values from a source tensor copied to specific indices of input tensor along the specified dimension.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/index-copy/index-copy.md:17]:
```suggestion
The **`.index_copy()`** function in PyTorch creates a new [tensor](https://www.codecademy.com/resources/docs/pytorch/tensors) with values from a source tensor copied to specific indices of the target tensor along the specified dimension.
```

File: content/pytorch/concepts/tensor-operations/terms/index-copy/index-copy.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+## Syntax
+
+```pseudo
+torch.index_copy(input_tensor, dim, index, source_tensor)
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/index-copy/index-copy.md:22]:
```suggestion
torch.index_copy_(dim, index, tensor)
```

File: content/pytorch/concepts/tensor-operations/terms/index-copy/index-copy.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
+---
+Title: 'index_copy()'
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/index-copy/index-copy.md:2]:
```suggestion
Title: '.index_copy_()'
```

File: content/pytorch/concepts/tensor-operations/terms/index-copy/index-copy.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+## Syntax
+
+```pseudo
+torch.index_copy(input_tensor, dim, index, source_tensor)
+```
+
+- `input_tensor`: The tensor to copy values into.
+- `dim`: The dimension along which to copy values.
+- `index`: A 1D tensor specifying the indices to copy to.
+- `source_tensor`: A tensor containing the values to be copied, matching dimension with the specified indices.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/index-copy/index-copy.md:28]: please change these based on the above syntax suggestion

File: content/pytorch/concepts/tensor-operations/terms/index-copy/index-copy.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+print("\nResult Tensor (Row Copy):\n", result)
+
+# Case-2: index_copy() along column
+index_col = torch.tensor([0, 2])
+source_tensor_col = torch.tensor([[1, 3], [1, 3], [1, 3]], dtype=torch.float)
+
+result = torch.index_copy(input_tensor, 1, index_col, source_tensor_col)
+
+print("\nResult Tensor (Column Copy):\n", result)
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/index-copy/index-copy.md:54]: can you make slight changes in this code, so that it fits with the `.index_copy_()`

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @SaiTeja-002, thank you for contributing to Codecademy Docs! 

The issue #5615 is about the `.index_copy()` in PyTorch, however, the `.index_copy()` is not a stable version, so can you just modify the entry to the stable version `.index_copy_()`?

I've left some suggestions in the entry as well, let me know when you make the changes 😄 
Thank you!

**Review by mamtawardhani — APPROVED** [review]
Thank you for making the changes @SaiTeja-002 😄 

The entry looks good for a second round of review! 🚀

**Review by Sriparno08 — APPROVED** [review]
Looks good, @SaiTeja-002!
